src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1828,9 +1828,24 @@ private function dumpValue(mixed $value, bool $interpolate = true): string
 
                         $attribute = sprintf('#[\Closure(%s)] ', $attribute);
                     }
-                    $this->addContainerRef = true;
 
-                    return sprintf("%sstatic function () use (\$containerRef)%s {\n            \$container = \$containerRef->get();\n\n            %s\n        }", $attribute, $returnedType, $code);
+                    if (str_contains($code, '$container')) {
+                        $this->addContainerRef = true;
+
+                        return sprintf(
+                            "%sstatic function () use (\$containerRef)%s {\n            \$container = \$containerRef->get();\n\n            %s\n        }",
```

Comment:
we could also use a short closure while at it :)

Reply:
Sounds good, will do in another PR!